src/useAccessibilityInfo.ts

```diff
@@ -1,43 +1,94 @@
 
-export function useAccessibilityInfo() {
-  const [reduceMotionEnabled, setReduceMotionEnabled] = useState(false)
-  const [screenReaderEnabled, setScreenReaderEnabled] = useState(false)
+const SUPPORTS_RN60_ACCESSIBILITY_INFO_API = !!(
+  AccessibilityInfo.isGrayscaleEnabled &&
+  AccessibilityInfo.isInvertColorsEnabled &&
+  AccessibilityInfo.isReduceMotionEnabled &&
+  AccessibilityInfo.isReduceTransparencyEnabled
+)
 
-  const handleReduceMotionChanged = (enabled: AccessibilityChangeEvent) =>
-    setReduceMotionEnabled(enabled)
-  const handleScreenReaderChanged = (enabled: AccessibilityChangeEvent) =>
-    setScreenReaderEnabled(enabled)
+type AccessibilityInfoStaticInitializers =
+  | 'isBoldTextEnabled'
+  | 'isScreenReaderEnabled'
+  | 'isGrayscaleEnabled'
+  | 'isInvertColorsEnabled'
+  | 'isReduceMotionEnabled'
+  | 'isReduceTransparencyEnabled'
+
+type AccessibilityEventToInfoStaticKeyMap = {
+  [K in AccessibilityChangeEventName]?: AccessibilityInfoStaticInitializers
+}
+
+const EVENT_NAME_TO_INITIALIZER: AccessibilityEventToInfoStaticKeyMap = {
+  boldTextChanged: 'isBoldTextEnabled',
+  screenReaderChanged: 'isScreenReaderEnabled',
+  grayscaleChanged: 'isGrayscaleEnabled',
+  invertColorsChanged: 'isInvertColorsEnabled',
+  reduceMotionChanged: 'isReduceMotionEnabled',
+  reduceTransparencyChanged: 'isReduceTransparencyEnabled',
+}
+
+function useAccessibilityStateListener(
+  eventName: AccessibilityChangeEventName,
+): boolean {
+  const [isEnabled, setIsEnabled] = useState(false)
 
   useEffect(() => {
-    AccessibilityInfo.isReduceMotionEnabled().then(handleReduceMotionChanged)
-    AccessibilityInfo.isScreenReaderEnabled().then(handleScreenReaderChanged)
-
-    AccessibilityInfo.addEventListener(
-      'reduceMotionChanged',
-      handleReduceMotionChanged as (event: AccessibilityEvent) => void,
-    )
-    AccessibilityInfo.addEventListener(
-      'screenReaderChanged',
-      handleScreenReaderChanged as (event: AccessibilityEvent) => void,
-    )
-
-    return () => {
-      AccessibilityInfo.removeEventListener(
-        'reduceMotionChanged',
-        handleReduceMotionChanged as (event: AccessibilityEvent) => void,
-      )
-      AccessibilityInfo.removeEventListener(
-        'screenReaderChanged',
-        handleScreenReaderChanged as (event: AccessibilityEvent) => void,
-      )
+    const initializerKey = EVENT_NAME_TO_INITIALIZER[eventName]
+
+    if (!initializerKey) {
+      return
+    }
+
+    AccessibilityInfo[initializerKey]().then(setIsEnabled)
+    AccessibilityInfo.addEventListener(eventName, setIsEnabled)
+
+    return () => AccessibilityInfo.removeEventListener(eventName, setIsEnabled)
+  }, [eventName])
+
+  return isEnabled
+}
+
+export function useAccessibilityInfo(): {
+  screenReaderEnabled: Boolean
+  boldTextEnabled: Boolean
+  grayscaleEnabled?: Boolean
+  invertColorsEnabled?: Boolean
+  reduceMotionEnabled?: Boolean
+  reduceTransparencyEnabled?: Boolean
```

Comment:
This function will return boolean scalars, not boxes

```suggestion
  screenReaderEnabled: boolean
  boldTextEnabled: boolean
  grayscaleEnabled?: boolean
  invertColorsEnabled?: boolean
  reduceMotionEnabled?: boolean
  reduceTransparencyEnabled?: boolean
```